compiler/front/cli_reporter.nim

```diff
@@ -1822,6 +1822,10 @@ proc reportBody*(conf: ConfigRef, r: SemReport): string =
     of rsemIllegalCustomPragma:
       result = "cannot attach a custom pragma to '$1'" % r.symstr
 
+    of rsemExternalIfaceNotAllowed:
+      result = "an external interface cannot be specified on a parameter or" &
+               " local 'var'/'let'"
```

Comment:
The whole terminology around the foreign-function-interface seems quite inconsistent too me, so I'm not sure.

For example, I'm not sure what is (or should be) considered an "external interface". Is it meant in the global sense, applying to the whole program? Or is it more meant in the local sense, only applying to symbols (the definition I went with when writing the error message)?

Reply:
I've been personally avoiding the module/compilation unit question(s) because I didn't feel like I've built up enough experience to formulate a passable answer. So yeah, it's also ambiguous for me whether we're talking about things on a per module basis or whole program (we do unity builds after all).

I think there is a bias towards it being on a module by module basis (compilation unit), so we can go with that for now until we have a more reasoned answer.

Reply:
Using "external interface" to refer to the external interface of the module, I think your suggested message fits better, so I replaced the previous one with it.